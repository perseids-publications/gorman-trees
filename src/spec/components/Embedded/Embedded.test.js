import React from 'react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { render, waitForElementToBeRemoved } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import config from '../../config.test.json';
import treebanks from '../../treebanks.test.json';

import Embedded from '../../../components/Embedded';

jest.mock('treebank-react');
jest.mock('alpheios-messaging');

const server = setupServer(
  rest.get(`${process.env.PUBLIC_URL}/xml/lysias-1-1-50.xml`, (req, res, ctx) => (
    res(ctx.delay(10), ctx.text(treebanks.lysias))
  )),
);

let sentenceCallbackValue;

beforeAll(() => server.listen());
beforeAll(() => {
  sentenceCallbackValue = global.sentenceCallbackValue;
  global.sentenceCallbackValue = {
    treebank: { $: {} },
    configuration: { deconstructPostag: () => {} },
  };
});
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());
afterAll(() => {
  global.sentenceCallbackValue = sentenceCallbackValue;
});

it('renders an embedded publication', async () => {
  const component = (
    <MemoryRouter initialEntries={['/embed/on-the-murder-of-eratosthenes-1-50/1']}>
      <Embedded config={config} />
    </MemoryRouter>
  );
  const { container, queryByText } = render(component);

  expect(container).toMatchSnapshot();

  await waitForElementToBeRemoved(() => queryByText('Loading...'));

  expect(container).toMatchSnapshot();
});

it('the API supports gotoSentence', async () => {
  const component = (
    <MemoryRouter initialEntries={['/embed/on-the-murder-of-eratosthenes-1-50/1']}>
      <Embedded config={config} />
    </MemoryRouter>
  );
  const { container, queryByText } = render(component);
  const request = { ID: 'test', body: { gotoSentence: { sentenceId: '5' } } };

  await waitForElementToBeRemoved(() => queryByText('Loading...'));
  global.sendRequestToMock(request, () => {});

  expect(container).toMatchSnapshot();
});

it('the API supports gotoSentence with highlighted words', async () => {
  const component = (
    <MemoryRouter initialEntries={['/embed/on-the-murder-of-eratosthenes-1-50/1']}>
      <Embedded config={config} />
    </MemoryRouter>
  );
  const { container, queryByText } = render(component);
  const request = { ID: 'test', body: { gotoSentence: { sentenceId: '5', wordIds: ['12'] } } };

  await waitForElementToBeRemoved(() => queryByText('Loading...'));
  global.sendRequestToMock(request, () => {});

  expect(container).toMatchSnapshot();
});

it('the API performs a no-op for refreshView', async () => {
  const component = (
    <MemoryRouter initialEntries={['/embed/on-the-murder-of-eratosthenes-1-50/1']}>
      <Embedded config={config} />
    </MemoryRouter>
  );
  const { container, queryByText } = render(component);
  const request = { ID: 'test', body: { refreshView: true } };

  await waitForElementToBeRemoved(() => queryByText('Loading...'));
  global.sendRequestToMock(request, () => {});

  expect(container).toMatchSnapshot();
});

it('renders 404 when publication not found', () => {
  const component = (
    <MemoryRouter initialEntries={['/embed/unknown']}>
      <Embedded config={config} />
    </MemoryRouter>
  );
  const { container } = render(component);

  expect(container).toMatchSnapshot();
});
