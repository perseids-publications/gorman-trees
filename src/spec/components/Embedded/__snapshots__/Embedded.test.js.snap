// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 404 when publication not found 1`] = `
<div>
  <div
    class="container pt-5"
  >
    <div
      class="row col-12 pt-5 pb-3"
    >
      <div
        class="col-12 text-center"
      >
        <h1>
          Error 404
        </h1>
      </div>
    </div>
    <div
      class="row col-12 pb-3"
    >
      <div
        class="col-12 text-center"
      >
        <h2>
          Publication not found
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders an embedded publication 1`] = `
<div>
  <div>
    <div>
      Loading...
    </div>
  </div>
</div>
`;

exports[`renders an embedded publication 2`] = `
<div>
  <div>
    <div
      class="treebankContainer"
    >
      <div
        class="text"
      >
        <div
          data-highlight=""
          data-id="1"
          data-treebank="1912971948"
        >
          Text
        </div>
      </div>
      <div
        class="graph"
      >
        <div
          data-highlight=""
          data-id="1"
          data-treebank="1912971948"
        >
          Graph
        </div>
      </div>
      <div
        data-highlight=""
        data-id="1"
        data-treebank="1912971948"
      >
        PartOfSpeech
      </div>
      <div
        class="links"
      >
        <p>
          <a
            href="https://www.example.com/on-the-murder-of-eratosthenes-1-50/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            Credits and more information
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`the API performs a no-op for refreshView 1`] = `
<div>
  <div>
    <div
      class="treebankContainer"
    >
      <div
        class="text"
      >
        <div
          data-highlight=""
          data-id="1"
          data-treebank="1912971948"
        >
          Text
        </div>
      </div>
      <div
        class="graph"
      >
        <div
          data-highlight=""
          data-id="1"
          data-treebank="1912971948"
        >
          Graph
        </div>
      </div>
      <div
        data-highlight=""
        data-id="1"
        data-treebank="1912971948"
      >
        PartOfSpeech
      </div>
      <div
        class="links"
      >
        <p>
          <a
            href="https://www.example.com/on-the-murder-of-eratosthenes-1-50/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            Credits and more information
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`the API supports gotoSentence 1`] = `
<div>
  <div>
    <div
      class="treebankContainer"
    >
      <div
        class="text"
      >
        <div
          data-highlight=""
          data-id="5"
          data-treebank="1912971948"
        >
          Text
        </div>
      </div>
      <div
        class="graph"
      >
        <div
          data-highlight=""
          data-id="5"
          data-treebank="1912971948"
        >
          Graph
        </div>
      </div>
      <div
        data-highlight=""
        data-id="5"
        data-treebank="1912971948"
      >
        PartOfSpeech
      </div>
      <div
        class="links"
      >
        <p>
          <a
            href="https://www.example.com/on-the-murder-of-eratosthenes-1-50/5"
            rel="noopener noreferrer"
            target="_blank"
          >
            Credits and more information
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`the API supports gotoSentence with highlighted words 1`] = `
<div>
  <div>
    <div
      class="treebankContainer"
    >
      <div
        class="text"
      >
        <div
          data-highlight="12"
          data-id="5"
          data-treebank="1912971948"
        >
          Text
        </div>
      </div>
      <div
        class="graph"
      >
        <div
          data-highlight="12"
          data-id="5"
          data-treebank="1912971948"
        >
          Graph
        </div>
      </div>
      <div
        data-highlight="12"
        data-id="5"
        data-treebank="1912971948"
      >
        PartOfSpeech
      </div>
      <div
        class="links"
      >
        <p>
          <a
            href="https://www.example.com/on-the-murder-of-eratosthenes-1-50/5"
            rel="noopener noreferrer"
            target="_blank"
          >
            Credits and more information
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
